src/ui-components/Address.tsx

```diff
@@ -56,39 +58,67 @@ const Address = ({ address, className, displayInline, disableIdenticon, extensio
 	const router = useRouter();
 	const [username, setUsername] = useState('');
 	const [kiltName, setKiltName] = useState('');
+	const[isAutoGeneratedUsername,setIsAutoGeneratedUsername] = useState(true);
 
 	const encoded_addr = address ? getEncodedAddress(address, network) || '' : '';
+	const FEATURE_RELEASE_DATE = dayjs('2023-06-12').toDate(); // Date from which we are sending custom username flag on web3 sign up.
+	const MANUAL_USERNAME_25_CHAR: string[] = ['marcosdanieldasilvasantos','masterquincyleroyharrison','sasha.fleur.polkaassembly'];
 
-	const fetchUsername = async () => {
+	const fetchUsername = async (isOnclick:boolean) => {
 		const substrateAddress = getSubstrateAddress(address);
-		if (!username && substrateAddress) {
+
+		if (substrateAddress) {
 			const { data, error } = await nextApiClientFetch<IGetProfileWithAddressResponse>(`api/v1/auth/data/profileWithAddress?address=${substrateAddress}`);
-			if (error) {
+			if (error|| !data || !data.username) {
 				console.error(error);
 				return;
 			}
-			if (data && data.username) {
-				setUsername(data.username);
+			setUsername(data.username);
+			if(isOnclick){
 				router.push(`/user/${data.username}`);
+				return;
+			}
+
+			if(MANUAL_USERNAME_25_CHAR.includes(data.username) || data.custom_username){
+				setIsAutoGeneratedUsername(false);
+				return;
+			}
+			if(data.username.length !== 25){
```

Comment:
this condition is not in the or yet, please do

Reply:
fixed